src/python/pants/backend/go/util_rules/build_pkg.py

```diff
@@ -320,23 +332,24 @@ async def _add_objects_to_archive(
     obj_file_paths: Iterable[str],
 ) -> ProcessResult:
     # Use `go tool asm` tool ID since `go tool pack` does not have a version argument.
-    asm_tool_id = await Get(GoSdkToolIDResult, GoSdkToolIDRequest("asm"))
-    pack_result = await Get(
-        ProcessResult,
-        GoSdkProcess(
-            input_digest=input_digest,
-            command=(
-                "tool",
-                "pack",
-                "r",
-                pkg_archive_path,
-                *obj_file_paths,
-            ),
-            env={
-                "__PANTS_GO_ASM_TOOL_ID": asm_tool_id.tool_id,
-            },
-            description="Link objects to Go package archive",
-            output_files=(pkg_archive_path,),
+    asm_tool_id = await compute_go_tool_id(GoSdkToolIDRequest("asm"))
+    pack_result = await execute_process_or_raise(
```

Comment:
The auto migrate put a couple of these as `fallible_to_exec_result_or_raise ` but I switched it to `execute_process_or_raise` which I saw elsewhere  https://github.com/pantsbuild/pants/pull/21375

Reply:
Yeah, you can do either - I think we'll do a mass re-name anyways eventually.